src/lib/domain/entities/post.js

```diff
@@ -46,11 +65,13 @@ export default class Post {
       slug: json.slug,
       description: json.description,
       content: json.content,
+      protectedContent: json.protected_posts ? json.protected_posts.content : null,
```

Comment:
you should create new entity to handle protected content called `postProtectedContent.js` that has `fromJson` function too.